single_kernel_mongo/managers/ldap.py

```diff
@@ -79,25 +82,37 @@ def on_ldap_ready(self, relation: Relation) -> None:
         """Runs when LDAP is ready."""
         self.assert_pass_hook_checks()
 
-        self.charm.status_manager.to_maintenance("Configuring LDAP")
+        self.charm.status_manager.set_and_share_status(MaintenanceStatus("Configuring LDAP"))
 
         ldap_data = self.ldap_requirer.consume_ldap_relation_data(relation=relation)
         if not ldap_data:
             logger.info("Waiting for LDAP data.")
             raise WaitingForLdapDataError("Waiting for LDAP data.")
 
         if not ldap_data.ldaps_urls:
-            logger.info("You must configure Glauth with LDAPS support")
+            logger.error("You must configure Glauth with LDAPS support")
             raise LDAPSNotEnabledError("LDAPS not enabled.")
 
         if self.charm.unit.is_leader():
             self.state.ldap.set_from(ldap_data)
 
+        self.restart_when_ready()
```

Comment:
Are there any dangers of not deferring when not ready?  (I assume no since the relation data would update and trigger the event calling this function)

Reply:
We specifically don't want to defer here:
1) we are waiting on two relations
2) restart when ready first gets the component status:
 * if there is no status, there is no integration so we don't restart
 * if there is an active status, then everything is integrated and we have checked that the connection works
 * Else, there is something blocking and we go in blocked state because this will not auto resolve.